### DISCUSSION TITLE
Contract methods getting failed. But in Polygon Explorer it succeeded.

### DISCUSSION THREAD

**Discussion by RajMazumder18110** [WalletConnect/walletconnect-monorepo#2490] [Developer Support]
Hi,
I am trying to build a react-native app. Whenever I am connected with walletconnect via Trust wallet and trying to call a smart contract method. The transaction either going into pending state or failed state. And it's happing on Polygon network. 

Contract is OK. Whenever I am calling the same method on Polygon Explorer via metamask it succeed.

Please help me.
It's look like this

https://github.com/WalletConnect/walletconnect-monorepo/assets/61340960/a0820198-f71d-4ce4-8ec6-c97e57e1f0ef

**Comment by xzilja** [comment]
Can you show some code / repository of how and which contracts you are calling?
